src/main/java/com/denizenscript/ddiscordbot/DiscordCommand.java

```diff
@@ -169,7 +235,114 @@ public void execute(ScriptEntry scriptEntry) throws CommandExecutionException {
                     dB.echoError(scriptEntry.getResidingQueue(), "Failed to message: unknown ID!");
                     return;
                 }
-                dDiscordBot.instance.connections.get(id.asString()).client.getChannelByID(channel.asLong()).sendMessage(message.asString());
+                RequestBuffer.request(() -> {
+                    dDiscordBot.instance.connections.get(id.asString()).client.getChannelByID(channel.asLong()).sendMessage(message.asString());
+                });
+                break;
+            case PRIVATE:
+                if (user == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no user given!");
+                    return;
+                }
+                if (message == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to message: no message given!");
+                    return;
+                }
+                if (!dDiscordBot.instance.connections.containsKey(id.asString())) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to message: unknown ID!");
+                    return;
+                }
+                IDiscordClient client_d = dDiscordBot.instance.connections.get(id.asString()).client;
+                RequestBuffer.request(() -> {
+                    client_d.getOrCreatePMChannel(client_d.getUserByID(user.asLong())).sendMessage(message.asString());
+                });
+                break;
+            case ROLE:
+                if (user == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no user given!");
+                    return;
+                }
+                if (guild == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no guild given!");
+                    return;
+                }
+                if (role == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no role given!");
+                    return;
+                }
+                if (message == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no action given!");
+                    return;
+                }
+                if (!dDiscordBot.instance.connections.containsKey(id.asString())) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: unknown ID!");
+                    return;
+                }
+                client_d = dDiscordBot.instance.connections.get(id.asString()).client;
+                IGuild guild_d = client_d.getGuildByID(guild.asLong());
+                switch (message.toString().toLowerCase()) {

```

Comment:
Why is the _message_ input used for add/remove? Give that a separate input.

Reply:
Any suggestion to what input it should have and prefix in the script argument?
` discord id:mybot role remove user:<def[user_id]> role:<def[role_id]> guild:<def[guild_id]>`
If I add `type` as an example
` discord id:mybot role type:remove user:<def[user_id]> role:<def[role_id]> guild:<def[guild_id]>`

Reply:
No prefix, just not treated as indistinct from the message.
Meta like `role (add/remove)` and give it a separate variable for tracking.
Or even do `addrole` and `removerole`